app/src/androidTest/java/de/test/antennapod/ui/PreferencesTest.java

```diff
@@ -35,36 +46,60 @@
 import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
 import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasDescendant;
 import static androidx.test.espresso.matcher.ViewMatchers.isChecked;
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
 import static androidx.test.espresso.matcher.ViewMatchers.withClassName;
+import static androidx.test.espresso.matcher.ViewMatchers.withHint;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static androidx.test.espresso.matcher.ViewMatchers.withText;
 import static de.test.antennapod.EspressoTestUtils.clickPreference;
 import static de.test.antennapod.EspressoTestUtils.waitForView;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.not;
 import static org.junit.Assert.assertTrue;
 
 @LargeTest
 public class PreferencesTest {
     private Resources res;
+    private Activity activity;
 
     @Rule
     public ActivityTestRule<PreferenceActivity> mActivityRule = new ActivityTestRule<>(PreferenceActivity.class, false, false);
 
+
     @Before
     public void setUp() {
         EspressoTestUtils.clearDatabase();
         EspressoTestUtils.clearPreferences();
         mActivityRule.launchActivity(new Intent());
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mActivityRule.getActivity());
+        activity = mActivityRule.getActivity();
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(activity);
         prefs.edit().putBoolean(UserPreferences.PREF_ENABLE_AUTODL, true).commit();
 
         res = mActivityRule.getActivity().getResources();
-        UserPreferences.init(mActivityRule.getActivity());
+        UserPreferences.init(activity);
+    }
+
+    @Test
+    public void testHideKeyboardWhenNavigatingUp() {
+        InputMethodManager imm = (InputMethodManager) mActivityRule.getActivity()
+                .getSystemService(Context.INPUT_METHOD_SERVICE);
+        onView(withId(R.id.recycler_view)).perform(
+                RecyclerViewActions.actionOnItem(
+                        allOf(hasDescendant(withHint(R.string.preference_search_hint))),
+                        click()));
+        onView(withClassName(new IsEqual<>(AppCompatImageButton.class.getName()))).perform(click());
+        Awaitility.await().atMost(1000, MILLISECONDS)
+                .until(()-> {
+                    View root = activity.findViewById(android.R.id.content);
+                    Rect visbileBounds = new Rect();
+                    root.getWindowVisibleDisplayFrame(visbileBounds);
+                    return root.getHeight() - visbileBounds.height() < 200;
```

Comment:
On my device, the test succeeds even if I comment out the fix and the keyboard is still open. So there seems to be something wrong here.

Reply:
> On my device, the test succeeds even if I comment out the fix and the keyboard is still open. So there seems to be something wrong here.

I'll take a look at it soon.

Reply:
@ByteHamster Update. Sorry i haven't been able to work on it. I'll probably get to it sometime this week.

Reply:
Hi, do you still have this PR in mind? :) If you don't manage to get the test to work, just remove it and we can merge it without the test :)

Reply:
I'll give it another shot today or tomorrow.

Reply:
The fix is now commented out. Also, the PR contains changes to other files that are (no longer) needed. Please revert the other changes, so that the displayed diff here on GitHub is clean. A clean git history helps to track down bugs more easily.